tools/getrealdeps.rb

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env ruby
+# getrealdeps for Chromebrew version 1.2
+# Author: Satadru Pramanik (satmandu) satadru at gmail dot com
+require 'fileutils'
+
+# Exit quickly if an invalid package name is given.
+if ARGV[0].nil? || ARGV[0].empty? || ARGV[0].include?('#')
+  puts 'Getrealdeps checks for the dependencies of a package.'
+  puts 'Dependencies are added if the package file is missing them.'
+  puts 'Usage: getrealdeps.rb <packagename>'
+  exit 1
+end
+
+pkg = ARGV[0].chomp('.rb')
+
+CREW_PREFIX = ENV['CREW_PREFIX'] || `crew const CREW_PREFIX`.split('=')[1].chomp
+CREW_LIB_PREFIX = ENV['CREW_LIB_PREFIX'] || `crew const CREW_LIB_PREFIX`.split('=')[1].chomp
+
+# Package needs to be installed for package filelist to be populated.
+unless File.exist? "#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist"
+  puts "Installing #{pkg} because it is not installed."
+  system("yes | crew install #{pkg}")
+end
+
+unless File.exist?("#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist")
+  puts "Package #{pkg} either does not exist or does not contain any libraries."
+  exit 1
+end
+
+# Speed up grep.
+ENV['LC_ALL'] = 'C'
+
+# Install grep if a functional local copy does not exist.
+if system('grep --version > /dev/null 2>&1')
+  GREP = 'grep'
+else
+  system('crew install grep')
+  GREP = "#{CREW_PREFIX}/bin/grep"
+end
+
+# Gawk is needed for adding dependencies.
+unless system('gawk -W version > /dev/null 2>&1')
+  puts "\nGawk is needed for adding dependencies to package files."
+  system('crew install gawk')
+end
+
+# Search for which packages have a needed library in CREW_LIB_PREFIX.
+# This is a subset of what crew whatprovides gives.
+def whatprovidesfxn(pkgdepslcl, pkg)
+  filelcl = if pkgdepslcl.include?(CREW_LIB_PREFIX)
+              `#{GREP} --exclude #{pkg}.filelist --exclude={"#{CREW_PREFIX}/etc/crew/meta/*_build.filelist"} "#{pkgdepslcl}$" "#{CREW_PREFIX}"/etc/crew/meta/*.filelist`
+            else
+              `#{GREP} --exclude #{pkg}.filelist --exclude={"#{CREW_PREFIX}/etc/crew/meta/*_build.filelist"} "^#{CREW_LIB_PREFIX}.*#{pkgdepslcl}$" "#{CREW_PREFIX}"/etc/crew/meta/*.filelist`
+            end
+  filelcl.gsub(/.filelist.*/, '').gsub(%r{.*/}, '').split("\n").uniq.join("\n").gsub(':', '')
+end
+
+# What files does a package provide.
+def crewfilesfxn(pkgname)
+  File.read("#{CREW_PREFIX}/etc/crew/meta/#{pkgname}.filelist")
+end
+
+pkgfiles = crewfilesfxn(pkg).split("\n").uniq
+
+FileUtils.rm_rf("/tmp/deps/#{pkg}")
+# Remove files we don't care about, such as man files and non-binaries.
+pkgfiles = pkgfiles.reject { |i| !File.file?(i.chomp) || !File.read(i.chomp, 4) == "\x7FELF" || i.include?('.zst') }
+
+# Use readelf to determine library dependencies, as
+# this doesn't almost run a program like using ldd would.
+pkgdepsfiles = pkgfiles.map do |i|
+  system("upx -d #{i} > /dev/null 2>&1") if File.exist?('/.dockerenv')
```

Comment:
Should there be a check to see if `upx` is installed first?

Reply:
That makes sense.

Reply:
Fixed